rotkehlchen/chain/evm/decoding/pendle/utils.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from typing import TYPE_CHECKING
+
+from eth_utils import to_normalized_address
+
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.constants.prices import ZERO_PRICE
+from rotkehlchen.errors.misc import NotERC20Conformant, RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.cache import (
+    globaldb_set_general_cache_values,
+    globaldb_set_unique_cache_value,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import CacheType, ChainID, ChecksumEvmAddress, Price
+from rotkehlchen.utils.network import request_get_dict
+
+from .constants import CPT_PENDLE
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_pendle_markets(chain: ChainID) -> None:
+    """Query Pendle markets for a given chain and cache their addresses."""
+    all_markets, all_sy_tokens = [], []
+    try:
+        for market_type in ['active', 'inactive']:  # query active and inactive markets
+            response_data = request_get_dict(url=f'https://api-v2.pendle.finance/core/v1/{chain.value}/markets/{market_type}')
```

Comment:
do you have a request session or the possibility to create a session where you call this? Since this is a for loop that can be quite heavy it might be a good idea to use a session to prevent duplicated efforts